src/subtitles/doc.md

```diff
@@ -0,0 +1,206 @@
+
+```
+moov.trak.mdia.minf.stbl.stts
+```
+- **Purpose**: Sample timing and duration
+- **Structure**: Array of {sample_count, sample_delta}
+- **Usage**: Calculate precise timestamps
+
+```
+moov.trak.mdia.minf.stbl.stsz
+```
+- **Purpose**: Sample sizes for byte range calculation
+- **Structure**: Array of sample sizes
+- **Usage**: Determine exact byte ranges to download
+
+```
+moov.trak.mdia.minf.stbl.stsc
+```
+- **Purpose**: Map samples to chunks
+- **Structure**: Array of {first_chunk, samples_per_chunk, sample_description_index}
+- **Usage**: Calculate chunk boundaries
+
+```
+moov.trak.mdia.minf.stbl.stco/co64
+```
+- **Purpose**: Chunk offset locations in file
+- **stco**: 32-bit offsets
+- **co64**: 64-bit offsets
+- **Usage**: Calculate absolute file positions
+
+### Subtitle Track Detection
+
+The system identifies subtitle tracks through multiple methods:
+
+1. **Handler Type Check**:
+   ```rust
+   // In hdlr box (bytes 8-12)
+   let handler_type = std::str::from_utf8(&hdlr_data[8..12]);
+   // "sbtl", "subt", "text" indicate subtitle tracks
+   ```
+
+2. **Media Info Check**:
+   ```rust
+   // In minf box
+   find_box(minf_data, "sbtl")  // Subtitle handler
+   find_box(minf_data, "subt")  // Subtitle handler
+   find_box(minf_data, "text")  // Text handler
+   ```
+
+## Extraction Process
+
+### 1. Format Detection
+- Skip MP3 files (no subtitles)
+- Support MP4 family formats (MP4, M4V, 3GP, 3G2, MOV)
+- Proceed with extraction for compatible formats
+
+### 2. Track Analysis
+- Locate `moov` box efficiently (8KB start/end search)
+- Navigate `moov → trak → mdia → minf → stbl` hierarchy
+- Identify subtitle tracks via `hdlr` box (handler type "sbtl"/"subt"/"text")
+
+### 3. Sample Table Processing
+- **STTS**: Sample timing and duration
+- **STSZ**: Sample sizes for byte range calculation
+- **STSC**: Sample-to-chunk mapping
+- **STCO/CO64**: Chunk offset locations
+
+### 4. Optimized Download
+- Calculate precise byte ranges for each subtitle sample
+- Group nearby ranges (gap < 4KB) to minimize HTTP requests
+- Download optimized chunks and parse subtitle data
+
+### 5. Format Parsing
+- Route to format-specific parsers based on codec type
+- Extract text content and apply default 2-second duration
+- Format timestamps in SRT standard (HH:MM:SS,mmm)
+
+## Supported Formats
+
+### TX3G (3GPP Timed Text)
+- **Structure**: 2-byte text length + UTF-8 text data
+- **Example**: `[00 1E] [text bytes...]` (30 bytes of text)
+
+### WebVTT
+- **Structure**: UTF-8 text with time markers
+- **Parsing**: Skip "WEBVTT" header, extract content
+
+### TTML (Timed Text Markup Language)
+- **Structure**: XML-based format
+- **Parsing**: Strip XML tags, extract text content
+
+### Generic
+- **Fallback**: UTF-8 first, then UTF-16
+- **Use**: Unknown or unsupported codec types
+
+## Usage Examples
+
+### Local File Extraction
+```rust
+use mediaparser::extract_subtitles;
+
+let entries = extract_subtitles("video.mp4").await;
```

Comment:
We need to support files with multiple subtitle tracks in different languages, and the data returned should include the language locale if that information is available.

I'm also not sure what our immediate use case will be for these extracted subtitles-- I wasn't there for that discussion. Search indexing? If you know @velocitysystems could you review this API and make sure it's returning all of the data we would need?